### ISSUE TITLE
Sentence does not have main clause 

### ISSUE THREAD

**Issue by cole-easton** [ethereum/ethereum-org-website#3397]
The sentence "Although we will be extremely judicious about making modifications to high-level constructs, for instance with the sharding roadmap, abstracting execution, with only data availability enshrined in consensus." found on https://ethereum.org/en/whitepaper/ does not have a main clause, and is thus grammatically invalid.

**Comment by wackerow** [comment]
👍🏻 Happy to accept a PR fix for this =)

**Comment by cole-easton** [comment]
> 👍🏻 Happy to accept a PR fix for this =)

I'm reading this because I'm not informed on the topic, so I'm not exactly sure what it's supposed to say.

**Comment by wackerow** [comment]
@cole-easton Fair enough 😂 Should've been more clear that was an open statement to anyone who would like to contribute. Appreciate you creating the issue 👍🏻

**Comment by sharon-wang** [comment]
TLDR: I had a little research journey to figure this out, since I'm also not well-versed in the topic and found the sentence confusing. I think what the sentence is trying to communicate is: when changes to high-level constructs are made, such as sharding, there's extensive discussion. The 'sharding roadmap' is an example where extensive discussion took place, ultimately resulting in the decision to "abstract execution, with only data availability enshrined in consensus."

I'll be opening a PR with a proposed rewording, based on my understanding :)

P.S. If anyone can point me to a definition/explanation of 'enshrine' in the context of Ethereum, I'd appreciate it as I still don't understand 😂 

---

I found https://github.com/ethereum/EIPs/issues/898#issuecomment-378587989, where this wording was introduced. It looks like the original wording was:
> 4. Agility - details of the Ethereum protocol are not set in stone. Although we will be extremely judicious about making modifications to high-level constructs such as the C-like language and the address system, computational tests later on in the development process may lead us to discover that certain modifications to the algorithm or scripting language will substantially improve scalability or security. If any such opportunities are found, we will exploit them.

Current wording for reference:
> 4. Agility: details of the Ethereum protocol are not set in stone. Although we will be extremely judicious about making modifications to high-level constructs, for instance with the [sharding roadmap](https://ethresear.ch/t/sharding-phase-1-spec/1407/), abstracting execution, with only data availability enshrined in consensus. Computational tests later on in the development process may lead us to discover that certain modifications, e.g. to the protocol architecture or to the Ethereum Virtual Machine (EVM), will substantially improve scalability or security. If any such opportunities are found, we will exploit them.

I'm also not informed on the topic, and was having a hard time understanding the 'sharding roadmap' example in the above text. Particularly, I'm not sure what "enshrined in consensus" means. With some searching, it seems that "enshrine" is a common term in the Ethereum community, which many people seem to already understand -- and as such, I did not come across a definition as usage was generally technical. Anyways, here's my research journey:

First, I found notes on [State and execution minimisation](https://notes.ethereum.org/@JustinDrake/BytY7RAFf) -- this includes the terms: enshrined, sharding, data availability, execution. Honestly, I couldn't make much sense of these notes with my limited knowledge of Ethereum, but at least it gave me ideas on other related terms to search for.

Then I came across the [diamond_drops repo](https://github.com/Drops-of-Diamond/diamond_drops), and in its README, it states:
> We're implementing sharding according to collaboration with Ethereum research at https://ethresear.ch and other teams, which includes tasks outlined in the projects and issues. In the first phase of the roadmap, we only enshrine data availabilty, without execution, thus abstracting execution, and enabling faster and simpler implementation, with each phase being useful.

With the above context from the README and with glancing over the mentioned [sharding roadmap](https://ethresear.ch/t/sharding-phase-1-spec/1407/), what I can deduce is that: when changes to high-level constructs are made, such as sharding, there's extensive discussion. I think this is what's meant to be communicated in "Although we will be extremely judicious about making modifications to high-level constructs, for instance with the sharding roadmap, abstracting execution, with only data availability enshrined in consensus." -- it's the careful deliberation that takes place which is being highlighted, and the "abstracting execution, with only data availability enshrined in consensus." part I believe is just some more detail on the sharding example (i.e. this part is beside the point).

**Comment by sharon-wang** [comment]
I've opened https://github.com/ethereum/ethereum-org-website/pull/3435. @cole-easton if you'd like, I would appreciate your thoughts on this rewording as I hope it clears things up for both of us!
